src/mds/objects.py

```diff
@@ -122,6 +137,96 @@ async def create_object(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.post("/objects/{guid:path}")
+async def create_object_for_id(
+    guid: str,
+    body: CreateObjForIdInput,
+    request: Request,
+    token: HTTPAuthorizationCredentials = Security(bearer),
+) -> JSONResponse:
+    """
+    Create object placeholder and attach metadata, return Upload url to the
+    user. A new GUID (new version of the provided GUID) will be created for
+    this object. The new record will have the same authz as the original one.
+
+    Args:
+        guid (str): indexd GUID or alias
+        body (CreateObjForIdInput): input body for create object for ID
+        request (Request): starlette request (which contains reference to FastAPI app)
+        token (HTTPAuthorizationCredentials, optional): bearer token
+    """
+    try:
+        # NOTE: token can be None if no Authorization header was provided, we expect
+        #       this to cause a downstream exception since it is invalid
+        token_claims = await access_token("user", "openid", purpose="access")(token)
+    except Exception as exc:
+        logger.error(exc, exc_info=True)
+        raise HTTPException(
+            HTTP_401_UNAUTHORIZED,
+            "Could not verify, parse, and/or validate scope from provided access token.",
+        )
+
+    # hit indexd's GUID/alias resolution endpoint to get the indexd did
+    try:
+        endpoint = config.INDEXING_SERVICE_ENDPOINT.rstrip("/") + f"/{guid}"
+        response = await request.app.async_client.get(endpoint)
+        response.raise_for_status()
+
+        # if the object is found in indexd, we can proceed
+        indexd_record = response.json()
+        indexd_did = indexd_record["did"]
+    except httpx.HTTPError as err:
+        logger.debug(err)
+        if err.response and err.response.status_code == 404:
+            msg = f"Could not find GUID or alias '{guid}' in indexd"
+            logger.debug(msg)
+            raise HTTPException(
+                HTTP_404_NOT_FOUND,
+                msg,
+            )
+        else:
+            msg = f"Unable to query indexd for GUID or alias '{guid}'"
+            logger.error(f"{msg}\nException:\n{err}", exc_info=True)
+            raise
+
+    file_name = body.file_name
+    aliases = body.aliases or []
+    metadata = body.metadata
+
+    metadata = metadata or {}
+
+    # get user id from token claims
+    uploader = token_claims.get("context", {}).get("user", {}).get("name")
```

Comment:
same note as before, the reason for using sub instead was that it's opaque and only we know how to map to the actual username (which may be protected information we don't want to expose via the MDS)